pkg/master/controller.go

```diff
@@ -200,6 +197,19 @@ func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP n
 			Type:            serviceType,
 		},
 	}
+	hash := strconv.FormatUint(hashService(svc), 10)
+	svc.Annotations = map[string]string{"kubernetes.io/config.hash": hash}
```

Comment:
Also @bgrant0607 will be happier if you change that annotation key to "apiserver.kubernetes.io/master-service-hash" or some such.